src/MCPClient/lib/clientScripts/verify_checksum.py

```diff
@@ -0,0 +1,276 @@
+system as part of a transfer, e.g. checksum.md5 in the transfer metadata
+folder.
+"""
+
+# This file is part of Archivematica.
+#
+# Copyright 2010-2017 Artefactual Systems Inc. <http://artefactual.com>
+#
+# Archivematica is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Archivematica is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Archivematica.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import print_function, unicode_literals
+
+import datetime
+import os
+import subprocess
+import sys
+
+import django
+from django.db import transaction
+
+django.setup()
+from main.models import File
+
+from custom_handlers import get_script_logger
+from databaseFunctions import insertIntoEvents
+
+
+def concurrent_instances():
+    """Function docstring. N.B. Delete me."""
+    return 1
+
+
+logger = get_script_logger("archivematica.mcp.client.compare_hashes")
+
+
+class NoHashCommandAvailable(Exception):
+    """Provide feedback to the user if the checksum command cannot be found
+    for the provided checksum file.
+    """
+
+
+class PREMISFailure(Exception):
+    """Provide feedback to the user if there is a problem writing PREMIS event
+    information to the database.
+    """
+
+
+class Hashsum(object):
+    """Class to capture various functions around calling Hashsum as a mechanism
+    for comparing user-supplied checksums in Archivematica.
+    """
+
+    COMMANDS = {
+        "metadata/checksum.md5": "md5sum",
+        "metadata/checksum.sha1": "sha1sum",
+        "metadata/checksum.sha256": "sha256sum",
+        "metadata/checksum.sha512": "sha512sum",
+        "metadata/checksum.b2": "b2sum",
+    }
+
+    OKAY_STRING = ": OK"
+    FAIL_STRING = ": FAILED"
+    ZERO_STRING = "no properly formatted"
+    IMPROPER_STRING = "improperly formatted"
+    FAILED_OPEN = ": FAILED open or read"
+    EXIT_NON_ZERO = "returned non-zero exit status 1"
+
+    def __init__(self, path, job=print):
+        """Class constructor."""
+        try:
+            self.COMMAND = self.COMMANDS[path]
+            self.job = job
+            self.command_called = None
+        except KeyError:
+            raise NoHashCommandAvailable()
+
+    def _call(self, *args):
+        """Make the call to Python subprocess and record the command being
+        called.
+        """
+        self.command_called = (self.COMMAND,) + args
+        return self._decode(subprocess.check_output(self.command_called))
+
+    def count_and_compare_lines(self, hashfile, objectsdir):
```

Comment:
If I understand the code paths correctly, `hashfile` here is the same `path` the constructor received? If that's the case why is it necessary to pass it again?

Reply:
Hmm, that's a good point. I think there was a reasonable origin for this, where you could create an object of type, `md5sum` maybe, and allowing other files to be validated against it, but it doesn't make sense in this context. And I didn't commit to that design. I will remove the arg and setup the attribute in the constructor. :+1: